drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -94,6 +95,13 @@ __attribute__((__section__(".unused_ram_start_marker")))
 static int unused_l2_sram_start_marker = 0xba0babce;
 #define UNUSED_L2_START_ALIGNED ROUND_UP(POINTER_TO_UINT(&unused_l2_sram_start_marker), \
 					 CONFIG_MM_DRV_PAGE_SIZE)
+#else
+/* If memory is not going to be powered off by the driver, just define
+ * the unused memory start as the end of the memory
+ */
+#define UNUSED_L2_START_ALIGNED ROUND_UP((L2_SRAM_BASE + L2_SRAM_SIZE), \
```

Comment:
Setting unused to HPSRAM top has more consequences:
- TLB driver blocks all map/unmap in range from HPSRAM start to UNUSED_L2_START_ALIGNED
this is because changing of any memory mapping in the area used by system (and managed during linkage) will mosltly lead to crash. And may also have a security impact

=> that means **no map/unmap is possible if MM_DRV_POWER_OFF_UNUSED_RAM = n**

From SOF point of view it does not matter, the option will always be "y"
But would suggest more accurate name of the option, like

MM_DRV_ALLOW_REMAPPING_OF_UNUSED_RAM
or - with oposite logic
MM_DRV_ADSP_DISABLE_REMAPPING_OF_VIRTUAL_MEMORY
 
with comment - if set to "y/n" memory not used during linkage will be turned off and may be used only when mapped to virtual memory space. If "n/y" - whole memory will stay powered up and mapped 1:1 to vrtual without possibility to remap.

Reply:
I'm not opposed to change the name, however it becomes clear that this "double bahaviour" warrants this configuration to be Intel ADSP only...

Reply:
Yes, TLB is a very specific device for ACE and its almost certain that is will never be used anywhere else

Reply:
Also, making this config more specific conflicts with previous review comment https://github.com/zephyrproject-rtos/zephyr/pull/51738#discussion_r1012253569 by @nashif - thoughts here?

Reply:
yes, have SoC names in options is just wrong and something we will have to change if someone else starts using that, so better keep it generic to what the option about and make it dependent on the SoC Kconfig.

Reply:
Any suggestions to the name than? Anything is fine for me as long as it describe what the option really does - in fact is disabling virtual memory mapping using TLB device leaving all the HPSRAM available to the linker

Reply:
I guess it should be fine if you left the MTL out of it and kept it to ADSP as being proposed above, do not really care...